controlplane/pkg/selector/selector.go

```diff
@@ -3,5 +3,5 @@ package selector
 import "github.com/ligato/networkservicemesh/controlplane/pkg/apis/registry"
 
 type Selector interface {
-	SelectEndpoint(ns *registry.NetworkService, networkServiceEndpoints []*registry.NetworkServiceEndpoint) *registry.NetworkServiceEndpoint
+	SelectEndpoint(ns *registry.NetworkService, nsLabels map[string]string, networkServiceEndpoints []*registry.NetworkServiceEndpoint) *registry.NetworkServiceEndpoint
```

Comment:
Perhaps we could just pass in the Connection and list of endpoints.  This gives us maximum flexibility.  We could for example choose to allow a Connection to specify an NSE by name.  Thoughts?

Reply:
I thought of that too. I'll play with the idea and see how it works.

Reply:
@edwarnicke I've implemented it and patch pushed here.